tests: net: convert normal test to use ztest

punitvara opened the pull request:
This patch convert net test cases to use ztest framework APIs
which makes output unified to other test cases.

Signed-off-by: Punit Vara <punit.vara@intel.com>

jukkar reviewed (changes requested):
Was there some reasoning why you split the stuff to main.c and then <component>.c? To me, this does not give much benefit.

punitvara committed:
tests: dhcpv4: convert normal test to use ztest

This patch convert net test cases to use ztest framework APIs
which makes output unified to other test cases.

Signed-off-by: Punit Vara <punit.vara@intel.com>

punitvara committed:
tests: neighbor: convert normal test to use ztest

This patch convert net test cases to use ztest framework APIs
which makes output unified to other test cases.

Signed-off-by: Punit Vara <punit.vara@intel.com>

punitvara commented:
@jukkar Done. Any problems to get it merge ?

nashif commented on tests/net/dhcpv4/src/dhcpv4.c:
```diff
@@ -527,7 +528,7 @@ static void receiver_cb(struct net_mgmt_event_callback *cb,
 }
 
-void main(void)
+void main_dhcp(void)
```
rename this, call this test_dhcp or whatever you are testing.

punitvara replied:
Done.

nashif commented on tests/net/dhcpv4/src/Makefile:
```diff
@@ -1,3 +1,4 @@
-obj-y = main.o
+obj-y = dhcpv4.o
```
keep it main.o

punitvara replied:
Done.

nashif commented on tests/net/neighbor/src/Makefile:
```diff
@@ -1,3 +1,4 @@
-obj-y = main.o
+obj-y = neighbor.o
```
keep it main.o

punitvara replied:
Done.

nashif commented on tests/net/neighbor/src/neighbor.c:
```diff
@@ -359,13 +358,17 @@ static bool run_tests(void)
 
 
-void main(void)
+void main_neighbor(void)
```
rename this, do not call this main_<something>, use the type of test you are doing as a name.

punitvara replied:
Got the idea. Thanks Anas

jukkar reviewed (approved):
LGTM
